sorc/fvcom_tools.fd/process_FVCOM.f90

```diff
@@ -73,6 +74,8 @@ program process_FVCOM
    character(len=180) :: wcstart
    character(len=180) :: inputFVCOMselStr
    character(len=180), dimension(:), allocatable :: args
+   integer            :: fv3_io_layout_y
```

Comment:
@hu5970 , From what I understand here, the domain is divided in the y-direction only and not x-direction for the mpi application, is this true? Also, this now requires the size of each subgrid in the y-direction to be passed to this program in the regional_workflow make_ics step. Is there a way to easily pass this automatically without user intervention?

Reply:
Yes, only y-direction is divided into subdomains for speeding up initialization. The x-direction can be divided but the suggest from FV3LAM experts is to divide Y (to set io_layout(2) >1).

In this code, the y-dimension for each subdomain is decided from reading the Y coordinate from grid_spec of each subdomain. No need user intervention. Also, this multiple subdomains only for warm start cycles. The cold start cycle still has single big domain for initial. That is why I moved the FVCOM run script out of make_ics and put it into prep step (a step prepare background for each data assimilation cycle).